### PR TITLE
Convert the `manifold-resource-product` component to GraphQL

### PR DESCRIPTION
**Pull request by Minivera**
Work is related to manifoldco/engineering#9192

## Reason for change

The container still partially uses REST, I am converting it in smaller steps to not create a giant untestable PR.

## Testing

Test using storybook, the component should work like it used to.

## Checklist

- [x] **CHANGELOG**: The **Unreleased** section of CHANGELOG was updated
- [x] **Prop changes**: [docs][docs] were updated
- [x] **Prop changes**: E2E tests were updated, testing from the highest level possible
- [x] **Platform testing**: If this change should be [tested against a platform][platform-testing], has it been?

[docs]: https://ui.sandbox.manifold.co
[platform-testing]: https://app.gitbook.com/@manifold/s/engineering/playbooks/testing-your-code-against-a-platform

### CODE REVIEW

File: src/components/manifold-resource-product/manifold-resource-product-happo.ts [hunk 1]
```diff
@@ -3,3 +3,28 @@ export const skeleton = () => {
...
+        label: 'product',
+        logoUrl: 'https://fillmurray.com/200/200',
+      },
+    },
+  };
+
+  const resourceProduct = document.createElement('manifold-resource-product');
+  resourceProduct.loading = false;
+  // TODO: Make good mocks or rethink the types, this ignore stops the compiler from wanting a complete resource
+  // @ts-ignore
```
Comment (dangodev) [inline src/components/manifold-resource-product/manifold-resource-product-happo.ts]: Why not `GraphqlResource as thing` instead?

File: src/components/manifold-resource-product/manifold-resource-product.tsx [hunk 2]
```diff
@@ -1,39 +1,50 @@
...
 import ResourceTunnel from '../../data/resource';
-import { Gateway } from '../../types/gateway';
+import { Product, Resource } from '../../types/graphql';
 import logger from '../../utils/logger';
 import loadMark from '../../utils/loadMark';
 
 @Component({ tag: 'manifold-resource-product' })
 export class ManifoldResourceProduct {
-  @Prop() data?: Gateway.Resource;
+  @Prop() gqlData?: Resource;
```
Comment (dangodev) [inline src/components/manifold-resource-product/manifold-resource-product.tsx]: Why rename this? (just curious)

File: src/components/manifold-resource-product/manifold-resource-product-happo.ts (hunk 1 above)
Comment (Minivera) [reply src/components/manifold-resource-product/manifold-resource-product-happo.ts]: Very, very good question 😅

File: src/components/manifold-resource-product/manifold-resource-product.tsx (hunk 2 above)
Comment (Minivera) [reply src/components/manifold-resource-product/manifold-resource-product.tsx]: The resource container injects two pieces of data into its state Tunnel, `data` which is the REST resource and `gqlData`, which is its GraphQL equivalent. I'm using `gqlData` only as a migration path and will rename everything when all the consumers are converted 😄

**Pull request merged**